### PR TITLE
[MKLDNN] Reusing once reordered convolution weights in test mode

### PR DESCRIPTION
**Pull request by kbinias**
Make sure for scoring weights in CONV are only converted once.

On ResNet-50, this change speeds up the whole topology on OMP_NUM_THREADS=1 and batch_size=1 by 15%

### CODE REVIEW

File: paddle/fluid/operators/conv_op.cc [hunk 1]
```diff
@@ -109,6 +109,7 @@ framework::OpKernelType ConvOp::GetExpectedKernelType(
 }
 
 void Conv2DOpMaker::Make() {
+  AddAttr<bool>("is_test", "").SetDefault(false);
```
Comment (Sand3r-) [inline paddle/fluid/operators/conv_op.cc:112]: It might be worth to move this line to where the other attributes are defined to keep the consistency. 
Also, consider adding the docs saying `(bool, default false) Only used in mkldnn kernel`.

**Review by Sand3r- — DISMISSED** [review]
Great job spotting that one @kbinias . Leaving here just a minor remark as to help keeping the code consistent.

File: paddle/fluid/operators/conv_op.cc (hunk 1 above)
Comment (kbinias) [reply paddle/fluid/operators/conv_op.cc:112]: This line is compatible with the content of the BatchNormOpMaker.Make() function.
Look at https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/batch_norm_op.cc source code.

**Review by Sand3r- — APPROVED** [review]

**Review by jczaja — APPROVED** [review]
LGTM
